thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/business/service/WrongAnswerService.java

```diff
@@ -84,7 +84,7 @@ private List<String> getWrongAnswers(Term rightAnswerTerm) {
                     break;
                 }
             }
-            if (!termService.existsOfCategory(potentialAnswer.word(), rightAnswerTerm.getCategory().getId())){
+            if (!termService.existsOfSearchPhrase(potentialAnswer.word(), rightAnswerTerm.getSearchPhrase().getId())){
```

Comment:
To be honest I am a little sceptic about this condition. Before, we had around 1000 words in the given `category`, now we have around 100 words in the given `SearchPhrase`. Sure, I understand that we would like not to have the name of a mountain in the question about rivers, but in my opinion we are rejecting to many words just by that. I think that the previous condition about `category` is a good compromise between being strict and trusting dataMuse. Feel free to challenge me with different opinion ;)

Reply:
Yeah, maybe you are right, especially in case of using multiple search_phrases for one "subcategory"